stdlib/Profile/src/Profile.jl

```diff
@@ -349,15 +349,18 @@ function getdict(data::Vector{UInt})
 end
 
 function getdict!(dict::LineInfoDict, data::Vector{UInt})
-    for ip in data
-        # Lookup is expensive, so do it only once per ip.
-        haskey(dict, UInt64(ip)) && continue
-        st = lookup(convert(Ptr{Cvoid}, ip))
-        # To correct line numbers for moving code, put it in the form expected by
-        # Base.update_stackframes_callback[]
-        stn = map(x->(x, 1), st)
-        try Base.invokelatest(Base.update_stackframes_callback[], stn) catch end
-        dict[UInt64(ip)] = map(first, stn)
+    # we don't want metadata here as we're just looking up ips
+    unique_data_itr = Iterators.unique(has_meta(data) ? strip_meta(data) : data)
+    foreach(ip -> dict[UInt64(ip)] = StackFrame[], unique_data_itr)
+    @sync for ip in unique_data_itr
+        Threads.@spawn begin
+            st = lookup(convert(Ptr{Cvoid}, ip))
+            # To correct line numbers for moving code, put it in the form expected by
+            # Base.update_stackframes_callback[]
+            stn = map(x->(x, 1), st)
+            try Base.invokelatest(Base.update_stackframes_callback[], stn) catch end
+            dict[UInt64(ip)] = map(first, stn)
```

Comment:
I always forget if `Dict` is threadsafe. I don't think it is?

Reply:
I thought it was if it's preallocated like it is here?

Reply:
At least experimentally they seem to be?
```
julia> dict = Dict{Int,Vector{Int}}();

julia> foreach(i -> dict[i] = Int[], 1:10_000)

julia> @sync for i in keys(dict)
           Threads.@spawn begin
               dict[i] = collect(1:i)
           end
       end

julia> for i in keys(dict)
           sum(dict[i]) == sum(1:i) || error()
       end

julia> 
```

Reply:
Ah yes as long as we are not reallocating...

Reply:
No, this is a very bad idea